tests/serverpod_test_server/test_integration/redis_cache_test.dart

```diff
@@ -125,4 +125,130 @@ void main() async {
       expect(value?.num, equals(1));
     });
   });
+
+  group('Given listener registered on channel', () {
+    const channelName = 'testChannel';
+    var messageSent = SimpleData(num: 1337);
+    SimpleData? messageReceived;
+    MessageCentralListenerCallback listener = (message) {
+      if (message is SimpleData) {
+        messageReceived = message;
+      }
+    };
+
+    setUp(
+      () async => session.messages.addListener(channelName, listener),
+    );
+    tearDown(() async {
+      session.messages.removeListener(channelName, listener);
+      messageReceived = null;
+    });
+
+    test('when global message is published to channel then message is received',
+        () async {
+      await session.messages.postMessage(
+        channelName,
+        messageSent,
+        global: true,
+      );
+
+      await Future.delayed(const Duration(milliseconds: 100));
```

Comment:
Suggestion: Modify this to await the message instead of a delay. Reasoning for that is that we cannot guarantee that this is a sufficient delay and may get flaky tests that fail when they shouldn't. With enough tests like this we may also suffer from long execution times.